File: python-wrapper/src/python/model/Replication.py
import json
from typing import Optional, List, Union, Dict
from model.embed.DatabaseSynchronization import DatabaseSynchronization
from dataclasses import field, dataclass

@dataclass
class Replication:
	id: str
	rev: Optional[str] = None
	deletion_date: Optional[int] = None
	name: Optional[str] = None
	context: Optional[str] = None
	database_synchronizations: List['DatabaseSynchronization'] = field(default_factory=list)

	def __serialize__(self) -> object:
		return {
			"id": self.id,
			"rev": self.rev,
			"deletionDate": self.deletion_date,
			"name": self.name,
			"context": self.context,
			"databaseSynchronizations": [x0.__serialize__() for x0 in self.database_synchronizations],
		}

	@classmethod
	def _deserialize(cls, data: Union[str, Dict[str, object]]) -> 'Replication':
		deserialized_dict: dict[str, object]
		if isinstance(data, str):
			deserialized_dict = json.loads(data)
		else:
			deserialized_dict = data
		return cls(
			id = deserialized_dict["id"],
			rev = deserialized_dict.get("rev"),
			deletion_date = deserialized_dict.get("deletionDate"),
			name = deserialized_dict.get("name"),
			context = deserialized_dict.get("context"),
			database_synchronizations = [DatabaseSynchronization._deserialize(x0) for x0 in deserialized_dict["databaseSynchronizations"]],
		)

__all__ = ['Replication']

File: cardinal-sdk/src/commonMain/kotlin/com/icure/cardinal/sdk/model/Replication.kt
package com.icure.cardinal.sdk.model

import com.icure.cardinal.sdk.model.base.Identifiable
import com.icure.cardinal.sdk.model.base.Named
import com.icure.cardinal.sdk.model.base.StoredDocument
import com.icure.cardinal.sdk.model.embed.DatabaseSynchronization
import com.icure.cardinal.sdk.utils.DefaultValue
import kotlinx.serialization.Serializable
import kotlin.Long
import kotlin.String
import kotlin.collections.List

@Serializable
public data class Replication(
	override val id: String,
	override val rev: String? = null,
	override val deletionDate: Long? = null,
	override val name: String? = null,
	public val context: String? = null,
	@DefaultValue("emptyList()")
	public val databaseSynchronizations: List<DatabaseSynchronization> = emptyList(),
) : StoredDocument, Identifiable<String>, Named
